commit 2fb85a37874c09fb22cd069f7c3c7fd6d8eb2f8f
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sat Dec 2 21:24:58 2023 +0100

    hurd: [!__USE_MISC] Do not #undef BSD macros in ioctls
    
    When e.g. including termios.h first and then sys/ioctl.h, without e.g.
    _BSD_SOURCE, the latter would #undef e.g. ECHO, without defining it.

diff --git a/sysdeps/mach/hurd/bits/ioctls.h b/sysdeps/mach/hurd/bits/ioctls.h
index 943d5e4287..1b2fd6b33f 100644
--- a/sysdeps/mach/hurd/bits/ioctls.h
+++ b/sysdeps/mach/hurd/bits/ioctls.h
@@ -25,6 +25,7 @@
 /* These macros are also defined in <bits/termios.h> (with numerically
    identical values) but this serves to shut up cpp's complaining. */
 
+#ifdef __USE_MISC
 #ifdef NL0
 # undef NL0
 #endif
@@ -82,6 +83,7 @@
 #ifdef NOFLSH
 # undef NOFLSH
 #endif
+#endif
 
 /* Hurd ioctl request are made up of several fields:
 
